substrafl/algorithms/pytorch/torch_fed_pca_algo.py

```diff
@@ -238,53 +234,61 @@ def train(
         # Create torch dataloader
         train_data_loader = torch.utils.data.DataLoader(train_dataset, batch_sampler=self._index_generator)
 
-        if shared_state is not None:
-            # The shared states are the model parameters.
-            # Hence we need to assign them to the previous local state parameters.
-            weight_manager.set_parameters(
-                model=self._model,
-                parameters=[torch.Tensor(shared_state.avg_parameters_update[0])],
-                with_batch_norm_parameters=False,
-            )
-
-        old_parameters = weight_manager.get_parameters(
-            model=self._model,
-            with_batch_norm_parameters=False,
-        )
-
         if self.local_mean is None:
-            # In round 1
-            new_parameters = self._compute_local_mean(
-                old_parameters=old_parameters,
+            # In round 1:
+            #   - Computation of the local mean
+            parameters_update = self._compute_local_mean(
                 train_data_loader=train_data_loader,
             )
 
         elif self.local_covmat is None:
-            # In round 2 we:
-            #   - Compute the local covariance matrix
-            #   - Initialize the weights for the subspace iteration method
-            #      and store them in old_parameters
-            new_parameters = self._compute_local_covmat(
-                old_parameters=old_parameters,
+            # In round 2:
+            #   - Computation of the local covariance matrix
+            #   - Initialization of the weights for the subspace iteration method
+            averaged_mean = torch.Tensor(shared_state.avg_parameters_update[0])
+
+            parameters_update = self._compute_local_covmat(
+                averaged_mean=averaged_mean,
                 train_data_loader=train_data_loader,
             )
-            # Assigning orthonormalized parameters
-            weight_manager.set_parameters(
-                model=self._model,
-                parameters=old_parameters,
-                with_batch_norm_parameters=False,
-            )
```

Comment:
this `weight_manager` update has been removed in the refactoring, was it intentional?

Reply:
Yep, it was not doing anything (I spent a lot of time on it to be sure ahah :) )